contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,419 @@
+pod's associated QoS class.
+
+### Guaranteed QoS
+
+A pod in this QoS class has its cgroup sandbox configured as follows:
+
+```
+pod<UID>/cpu.shares = sum(pod.spec.containers.resources.requests[cpu])
+pod<UID>/cpu.cfs_quota_us = sum(pod.spec.containers.resources.limits[cpu])
+pod<UID>/memory.limit_in_bytes = sum(pod.spec.containers.resources.limits[memory])
+```
 
+### Burstable QoS
 
-__Note__: The cgroup-root flag would allow the user to configure the root of the QoS cgroup hierarchy. Hence cgroup-root would be redefined as the root of QoS cgroup hierarchy and not containers.
+A pod in this QoS class has its cgroup sandbox configured as follows:
 
 ```
-/PodUID/cpu.quota = cpu limit of Pod  
-/PodUID/cpu.shares = cpu request of Pod  
-/PodUID/memory.limit_in_bytes = memory limit of Pod
+pod<UID>/cpu.shares = sum(pod.spec.containers.resources.requests[cpu])
+```
+
+If all containers in the pod specify a cpu limit:
+
+```
+pod<UID>/cpu.cfs_quota_us = sum(pod.spec.containers.resources.limits[cpu])
+```
+
+Finally, if all containers in the pod specify a memory limit:
+
+```
+pod<UID>/memory.limit_in_bytes = sum(pod.spec.containers.resources.limits[memory])
+```
+
+Note: This design enables containers in a pod to optionally share slack compute resources.
+
+### BestEffort QoS
+
+A pod in this QoS class has its cgroup sandbox configured as follows:
+
 ```
+pod<UID>/cpu.shares = 2
+```
+
+## QoS level cgroups
+
+The `kubelet` defines a `--cgroup-root` flag that is used to specify the `ROOT`
+node in the cgroup hierarchy below which the `kubelet` should manange individual
+cgroup sandboxes.
+
+The `ROOT` cgroup sandbox is used to parent all pod sandboxes that are in
+the Guaranteed QoS class.  By definition, pods in this class have cpu and 
+memory limits specified that are equivalent to their requests so the pod 
+level cgroup sandbox confines resource consumption without the need of an
+additional cgroup sandbox for the tier.
+
+When the `kubelet` launches, it will ensure a `Burstable` cgroup sandbox
+and a `BestEffort` cgroup sandbox exist as children of `ROOT`.  These cgroup
+sandboxes will parent pod level cgroups in those associated QoS classes.
+
+The `kubelet` highly prioritizes resource utilization, and thus
+allows BestEffort and Burstable pods to potentially consume as many
+resources that are presently available on the node.
+
+For compressible resources, this prioritization scheme has little impact.
+For example, CPU time is proportioned dynamically when there is contention
+using CFS shares that ensure minimum requests are satisfied.
+
+For incompressible resources, this prioritization scheme can inhibit the
+ability of a pod to have its requests satisfied.  For example, a Guaranteed
+pods memory request may not be satisfied if there are active BestEffort
+pods consuming all available memory.
+
+The `kubelet` will support a flag `experimental-qos-reserve-requests` that
```

Comment:
by definition, we are reserving for G pods in the same manner we are reserving for node allocatable top level cgroup.  both have a memory limit set equal to their request.  i agree with @sjenning .  i think overcommit-percentage means something like a physical cpu/memory to virtual cpu/memory scalar that would skew what the node reports as allocatable back to the scheduler.  this is precisely not that.  this is attempting to reserve for the qos tier in the same manner we are reserving for the kubepods cgroup sandbox one level up in the hierarchy.  it's also experimental in 1.6, so i am not sure how much we want to bikeshed on the name.